server/command.js

```diff
@@ -0,0 +1,243 @@
+var $ = function(str) {
+  // Parse the string to extract a reference.
+  // TODO: Support notations starting with @roomname and ~username.
+  var m = str.match(/^\s*(\$|me|~|here|@)(.*)\s*$/);
+  if (!m) {
+    return undefined;
+  }
+  var root = m[1];
+  var suffix = m[2];
+  if (root === '$') {
+    root = $;
+  } else if (root === '~' || root === 'me') {
+    root = user;
+  } else if (root === '@' || root === 'here') {
+    root = user.location;
+  } else {
+    throw Error("Can't happen.  Regex is too liberal.");
+  }
+  if (!suffix) {
+    return root;
+  }
+  // Clear no-longer needed local variables so they don't leak into the eval.
+  str = undefined;
+  m = undefined;
+  return eval('(root)' + suffix);
+};
+
+
+$.utils = {};
+$.utils.command = {};
```

Comment:
A flattened $ hierarchy would have a thousand names at the root level and be non-explorable.